packages/contracts-bedrock/scripts/periphery/drippie/DrippieConfig.s.sol

```diff
@@ -0,0 +1,149 @@
+        // checkparams as a tuple which can't be parsed in a generic way (since Solidity does not
+        // support generics). As a result, we first parse the array as a simplified struct that
+        // only includes the first two fields so that we can determine the number of drips. We then
+        // iterate over the array and parse the full struct for each drip somewhat manually.
+        CoreDripConfig[] memory corecfg = abi.decode(stdJson.parseRaw(_json, "$.drips"), (CoreDripConfig[]));
+        console.log("DrippieConfig: found %d drips", corecfg.length);
+
+        // Iterate and parse all of the drips.
+        for (uint256 i = 0; i < corecfg.length; i++) {
+            // Log so we know what's being loaded.
+            string memory name = corecfg[i].name;
+            console.log("DrippieConfig: attempting to load config for %s", name);
+
+            // Make sure the dripcheck is deployed.
+            string memory dripcheck = corecfg[i].dripcheck;
+            console.log("DrippieConfig: attempting to get address for %s", dripcheck);
+            mustGetAddress(dripcheck);
+
+            // Generate the base JSON path string.
+            string memory p = string.concat("$.drips[", vm.toString(i), "]");
+
+            // Parse all the easy stuff first.
+            console.log("DrippieConfig: attempting to load core configuration for %s", name);
+            FullDripConfig memory dripcfg = FullDripConfig({
+                name: name,
+                dripcheck: dripcheck,
+                checkparams: bytes(string("__WE_WILL_PARSE_THIS_NEXT__")),
+                recipient: stdJson.readAddress(_json, string.concat(p, ".03__recipient")),
+                value: stdJson.readUint(_json, string.concat(p, ".04__value")),
+                interval: stdJson.readUint(_json, string.concat(p, ".05__interval")),
+                data: stdJson.parseRaw(_json, string.concat(p, ".06__data"))
+            });
+
+            // Load the checkparams as bytes.
+            bytes memory checkparams = stdJson.parseRaw(_json, string.concat(p, ".02__checkparams"));
+
+            // Determine if the parameters are decodable.
+            console.log("DrippieConfig: attempting to decode check parameters for %s", name);
+            if (strcmp(dripcheck, "CheckBalanceLow")) {
+                abi.decode(checkparams, (CheckBalanceLow.Params));
+            } else if (strcmp(dripcheck, "CheckGelatoLow")) {
+                abi.decode(checkparams, (CheckGelatoLow.Params));
+            } else if (strcmp(dripcheck, "CheckSecrets")) {
+                abi.decode(checkparams, (CheckSecrets.Params));
+            } else if (strcmp(dripcheck, "CheckTrue")) {
+                // No parameters to decode.
+            } else {
+                console.log("ERROR: unknown drip configuration %s", dripcheck);
+                revert UnknownDripCheck(dripcheck);
+            }
+
+            // Ok we're good to go.
+            dripcfg.checkparams = checkparams;
```

Comment:
Would prefer if we just parsed the `checkparams` before constructing `dripcfg`, to remove the need for the intermediate placeholder: `bytes(string("__WE_WILL_PARSE_THIS_NEXT__"))`

Doesn't look like the other fields within `dripcfg` (except `name`, which has a copy external to the struct) is needed, so can just re-order.